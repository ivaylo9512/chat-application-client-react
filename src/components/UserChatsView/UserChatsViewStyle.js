import styled from "styled-components";

export const Container = styled.div`
    width: 100%;
    flex: 1;
    display: flex;
    flex-direction: column;
    overflow-y: scroll;
    padding-bottom: 2vh;
`