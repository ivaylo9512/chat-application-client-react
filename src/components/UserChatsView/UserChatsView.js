import { resetUserChatsState, getUserChatsQuery, userChatsRequest, getUserChatsState, getUserChats, setCurrentUserChats } from 'app/slices/userChatsSlice';
import Pagination from 'components/Pagination/Pagination';
import UserChatsList from 'components/UserChatsList/UserChatsList';
import Form from 'components/Form/Form';
import { useSelector } from 'react-redux';
import { Container } from './UserChatsViewStyle';

const UserChatsView = () => {
    return(
        <>
            <Container>
                <UserChatsList />
                <Pagination selector={getUserChatsState} setData={setCurrentUserChats} getData={userChatsRequest}/>
            </Container>
            <Form action={userChatsRequest} resetState={resetUserChatsState} selector={getUserChatsQuery} placeholder={'search chat'} />
        </>
    )
}
export default UserChatsView