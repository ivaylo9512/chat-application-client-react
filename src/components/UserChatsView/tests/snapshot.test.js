import UserChatsView from 'components/UserChatsView/UserChatsView';
import { shallow } from 'enzyme';
import * as redux from 'react-redux'
describe('UserChatsView snapshot tests', () => {    
    let wrapper;

    const createWrapper = (state) => {
        jest.spyOn(redux, 'useSelector').mockReturnValue(state);
        jest.spyOn(redux, 'useDispatch').mockReturnValue(jest.fn());
     
        wrapper = shallow(
            <UserChatsView />
        )
    }

    it('should match snapshot', () => {
        createWrapper({ isLoading: false });

        expect(wrapper).toMatchSnapshot();
    })

    it('should match snapshot with loading request', () => {
        createWrapper({ isLoading: true });

        expect(wrapper).toMatchSnapshot();
    })

    it('should match snapshot with error', () => {
        createWrapper({ isLoading: false, error: 'Unavailable' });

        expect(wrapper).toMatchSnapshot();
    })
})