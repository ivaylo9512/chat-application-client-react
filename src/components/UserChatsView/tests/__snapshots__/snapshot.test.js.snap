// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserChatsView snapshot tests should match snapshot 1`] = `
<Fragment>
  <styled.div>
    <UserChatsList />
    <Pagination
      getData={[Function]}
      pagesPerSlide={5}
      selector={[Function]}
      setData={[Function]}
    />
  </styled.div>
</Fragment>
`;

exports[`UserChatsView snapshot tests should match snapshot with error 1`] = `
<Fragment>
  <styled.div>
    <UserChatsList />
    <div
      data-testid="error"
    >
      Unavailable
    </div>
    <Pagination
      getData={[Function]}
      pagesPerSlide={5}
      selector={[Function]}
      setData={[Function]}
    />
  </styled.div>
</Fragment>
`;

exports[`UserChatsView snapshot tests should match snapshot with loading request 1`] = `
<Fragment>
  <styled.div>
    <UserChatsList />
    <LoadingIndicator />
  </styled.div>
</Fragment>
`;
